client/src/containers/Home.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import './Home.less';
+import { useState } from 'react';
+import CreateContact from '../components/CreateContact';
+import DB from '../services/dbService';
+import InputTime from '../components/InputTime';
+import CallPaneReceiver from '../components/CallPaneReceiver';
+import { listenForIncomingCall } from '../services/WebSocketService'
+import CallExpired from '../components/CallExpired'
+
+export const RenderContext = React.createContext(null);
+export const CallerContext = React.createContext(null);
+
+function Home() {
+
+  const [createContact, setCreateContact] = useState(false)
+  const [selectContact, setSelectContact] = useState('')
+  const [goToTimeInput, setGoToTimeInput] = useState(false)
+  const [incomingCall, setIncomingCall] = useState(false)
+  const [callExpired, setCallExpired] = useState(false)
+  const [incomingTimeData, setIncomingTimeData] = useState({})
+
+  const setIncomingCallFlag = (timeData) => {
+    setIncomingCall(true)
+    setIncomingTimeData(timeData)
+  }
+
+  listenForIncomingCall(setIncomingCallFlag)
+
+  // const addAContact = (flag) => {
+  //   setCreateContact(flag)
+  // }
+
+  const selectContactToCall = (contact) => {
+    setSelectContact(contact)
+    setGoToTimeInput(true)
+  }
+
+  const callHasExpired = (flag) => {
+    setCallExpired(flag)
+  }
+  
+  if (callExpired) {
+    console.log('yo')
+    return (
+      <div>
+        <CallExpired />
+      </div>
+    )
+  } else if (incomingCall && incomingTimeData) {
+    return (
+      <div>
+        <CallerContext.Provider >
```

Comment:
I think it's not a good idea to conditionally render a Provider. One cool thing about Contexts is that you can leave it there, if you consume it nice, if not also nice.

Also in this particular case your setting a provider with no value.

Reply:
The provider is no longer used so have removed.